ksql-benchmark/src/main/java/io/confluent/ksql/benchmark/SerdeBenchmark.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.benchmark;
+
+import static io.confluent.ksql.datagen.DataGenSchemaUtil.getOptionalSchema;
+
+import io.confluent.avro.random.generator.Generator;
+import io.confluent.connect.avro.AvroData;
+import io.confluent.kafka.schemaregistry.client.MockSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.datagen.RowGenerator;
+import io.confluent.ksql.datagen.SessionManager;
+import io.confluent.ksql.serde.avro.KsqlAvroTopicSerDe;
+import io.confluent.ksql.serde.json.KsqlJsonTopicSerDe;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.Pair;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.Schema;
+
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+public final class SerdeBenchmark {
+
+  private static final Path SCHEMA_DIR = Paths.get("schemas");
+  private static final String topicName = "serde_benchmark";
+
+  private SerdeBenchmark() {
+  }
+
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  @Warmup(iterations = 5, time = 10)
+  @Measurement(iterations = 5, time = 10)
+  @Threads(4)
+  public abstract static class AbstractBenchmark {
+
+    @State(Scope.Thread)
+    public static class SchemaAndGenericRowState {
+      org.apache.kafka.connect.data.Schema schema;
+      GenericRow row;
+
+      @Param({"impressions", "metrics"})
```

Comment:
Ideally we'd like to scan the `schemas/` directory to find schemas automatically, rather than hard-coding schema names here. Working on this change now, but wanted to open this PR for review before getting the change in.

Reply:
Update on scanning the `schemas/` directory to find schemas rather than hard-coding the list here: I wasn't able to do this because the `Param` annotation requires a constant value, and using a method to scan the directory at runtime violates this requirement. Do let me know if I've missed something!